pyteal/ast/assert_.py

```diff
@@ -12,22 +13,56 @@
 class Assert(Expr):
     """A control flow expression to verify that a condition is true."""
 
-    def __init__(self, cond: Expr, *additional_conds: Expr) -> None:
+    def __init__(
+        self,
+        cond: Expr | tuple[Expr, Optional[str]],
+        *additional_conds: Expr | tuple[Expr, Optional[str]]
+    ) -> None:
         """Create an assert statement that raises an error if the condition is false.
 
         Args:
             cond: The condition to check. Must evaluate to a uint64.
             *additional_conds: Additional conditions to check. Must evaluate to uint64.
         """
         super().__init__()
+
+        self.cond: list[Expr] = []
+        self.comments: list[Optional[str]] = []
+
+        if type(cond) is tuple:
+            cond, comment = cond
+            self.comments.append(comment)
+            self.cond.append(cond)
+
+            self.comment = comment
+        else:
+            self.cond.append(cast(Expr, cond))
+            # Keep matching indicies for expr/comment
+            self.comments.append(None)
+
         require_type(cond, TealType.uint64)
+
         for cond_single in additional_conds:
+            if type(cond_single) is tuple:
+                cond_single, cond_single_comment = cond_single
+                self.comments.append(cond_single_comment)
+                self.cond.append(cond_single)
+            else:
+                self.comments.append(None)
+                self.cond.append(cast(Expr, cond_single))
+
             require_type(cond_single, TealType.uint64)
-        self.cond = [cond] + list(additional_conds)
 
     def __teal__(self, options: "CompileOptions"):
         if len(self.cond) > 1:
-            asserts = [Assert(cond) for cond in self.cond]
+            asserts = []
+            for idx, cond in enumerate(self.cond):
+                a = Assert(cond)
+                a.trace = cond.trace
+                if self.comments[idx] is not None:
+                    a = cast(Assert, Comment(cast(str, self.comments[idx]), a))
```

Comment:
I don't like this -- how about declare `asserts: list[Expr] = []` instead of a bad cast?